nixos/modules/services/display-managers/greetd.nix

```diff
@@ -0,0 +1,105 @@
+{ config, lib, pkgs, ... }:
+with lib;
+
+let
+  cfg = config.services.greetd;
+  tty = "tty${toString cfg.vt}";
+  settingsFormat = pkgs.formats.toml {};
+in
+{
+  options.services.greetd = {
+    enable = mkEnableOption "greetd";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.greetd.greetd;
+      defaultText = "pkgs.greetd.greetd";
+      description = "The greetd package that should be used.";
+    };
+
+    settings = mkOption {
+      type = settingsFormat.type;
+      example = literalExample ''
+        {
+          default_session = {
+            command = "''${pkgs.greetd.greetd}/bin/agreety --cmd sway";
+          };
+        }
+      '';
+      description = ''
+        greetd configuration (<link xlink:href="https://man.sr.ht/~kennylevinsen/greetd/">documentation</link>)
+        as a Nix attribute set.
+      '';
+    };
+
+    vt = mkOption  {
+      type = types.int;
+      default = 1;
+      description = ''
+        The virtual console (tty) that greetd should use. This option also disables getty on that tty.
+      '';
+    };
+
+    restart = mkOption {
```

Comment:
Does it make more sense to automatically do this for the sysadmin?

`serviceConfig.Restart = mkDefault cfg.settings.initial_session or false;`

Reply:
I like the general idea but dislike the hidden behavior, so I left the `restart`-option in but set it's default accordingly. Is that an acceptable solution?

Reply:
I was really asking a question, not implying the answer... hence any solution, including the original, is an acceptable solution to me.

😄

Reply:
I guess that wasn't phrased well, I just wanted to ask for feedback on leaving the option :)
Anyway, I'm happy with the current version. Adding the default reduces the chance for accidental insecure deployments.